### PR TITLE
Allow pass config items through command line.

### PR DESCRIPTION
**Pull request by haosdent**

### CODE REVIEW

**Comment by haosdent** [comment]
ping @ideawu

**Comment by ideawu** [comment]
Hi, there is no need to do such thing, just edit the config file would be better.

**Comment by haosdent** [comment]
Hi, because we need use ssdb in docker. Pass configuration through environment variables would be better. So that I don't need to build docker image everytime when I change configuration

File: src/util/app.cpp [hunk 1]
```diff
@@ -53,27 +54,50 @@ void Application::parse_args(int argc, char **argv){
...
 	if(!conf){
 		fprintf(stderr, "error loading conf file: '%s'\n", app_args.conf_file.c_str());
 		exit(1);
 	}
+
+	for(std::vector<std::string>::const_iterator iter = external_config_items.begin();
+		iter != external_config_items.end();
+		++iter){
+		std::string external_config_item = *iter;
+		size_t split_point = external_config_item.find_first_of(":=");
```
Comment (ideawu) [inline src/util/app.cpp]: `:=` or `=`?

File: src/util/app.cpp [hunk 2]
```diff
@@ -53,27 +54,50 @@ void Application::parse_args(int argc, char **argv){
...
+		if(split_point < 0 && split_point >= external_config_item.length()){
```
Comment (ideawu) [inline src/util/app.cpp]: use string::npos,
```
if(split_point != std::string::npos){
```

File: src/util/app.cpp (hunk 1 above)
Comment (haosdent) [inline src/util/app.cpp]: Oh, k:v or k=v are legal here.

File: src/util/app.cpp (hunk 2 above)
Comment (haosdent) [inline src/util/app.cpp]: Thx! Let me update.

File: src/util/config.cpp [hunk 3]
```diff
@@ -13,23 +13,28 @@ int is_kv_seperator(int ch){
 }
```
Comment (ideawu) [inline src/util/config.cpp]: Please do not edit this file.

File: src/util/app.cpp (hunk 1 above)
Comment (ideawu) [inline src/util/app.cpp]: Yes, you are right.

File: src/util/config.cpp (hunk 3 above)
Comment (ideawu) [inline src/util/config.cpp]: Make the fewest modification as possible.

**Comment by ideawu** [comment]
Well, I will merge this request after you update the code according to my notes.

File: src/util/app.cpp [hunk 4]
```diff
@@ -53,27 +54,50 @@ void Application::parse_args(int argc, char **argv){
...
 
-	if(app_args.conf_file.empty()){
+	if(app_args.conf_file.empty() && external_config_items.empty()){
 		usage(argc, argv);
 		exit(1);
 	}
 }
 
 void Application::init(){
-	if(!is_file(app_args.conf_file.c_str())){
```
Comment (ideawu) [inline src/util/app.cpp]: test if conf_file is empty here, not making any change to config.cpp:

```
	if(app_args.conf_file.empty()){
		conf = new Config();
	}else{
		if(!is_file(app_args.conf_file.c_str())){
			fprintf(stderr, "'%s' is not a file or not exists!\n", app_args.conf_file.c_str());
			exit(1);
		}
		conf = Config::load(app_args.conf_file.c_str());
	}
```

```

**Pull request closed without merging**